crates/router/src/connector/bluesnap/transformers.rs

```diff
@@ -63,6 +70,21 @@ pub struct BluesnapPaymentsRequest {
     transaction_fraud_info: Option<TransactionFraudInfo>,
     card_holder_info: Option<BluesnapCardHolderInfo>,
     merchant_transaction_id: Option<String>,
+    transaction_meta_data: Option<BluesnapMetadata>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BluesnapMetadata {
+    meta_data: Vec<RequestMetadata>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RequestMetadata {
+    meta_key: Option<String>,
+    meta_value: Option<String>,
+    is_visible: Option<String>,
```

Comment:
Why do we need to keep these fields as optional? RequestMetadata should be instructed only incase of payment metadata is present

Reply:
These fields are optional at the connector end. Hence made them optional in `RequestMetadata` struct